### DISCUSSION TITLE
loc_loss 误差这么小 为什么推理   跟踪突变这么大  718-》837

### DISCUSSION THREAD

**Discussion by wavelet2008** [STVIR/pysot#475] [Q&A]
2020-11-19 19:12:20,716-rk0-train.py#196] epoch: 14
[2020-11-19 19:12:20,716-rk0-train.py#201] epoch 14 lr 0.0009026601969102485
[2020-11-19 19:12:20,716-rk0-train.py#201] epoch 14 lr 0.0009026601969102485
[2020-11-19 19:12:25,057-rk0-train.py#249] Epoch: [14][15/625] lr: 0.001805
	batch_time: 0.311095 (0.310400)	data_time: 0.241359 (0.240799)
	cls_loss: 0.002425 (0.008269)	loc_loss: 0.116004 (0.124908)

0.93641865
 x1=718  , y1=287  x2=76  , y2=109 
269.2549228668213milliseconds

0.93656695
 x1=837  , y1=337  x2=89  , y2=136

**Comment by ZhiyuanChen** [comment]
所以你的问题是？

**Reply by wavelet2008** [reply]
我加上自己数据，训练效果很差，跟踪效果很差，完全不对，rpn_head.loc.1.running_var 系数比大神提供的差距比较多。目前训练数据ytbb 中的train0000-train0019 加上自己数据，用 siamrpn_alex_dwxcorr 训练的  求大神指点

**Reply by ZhiyuanChen** [reply]
我无法从你所提供的信息中获得任何有价值的内容，比如说你的数据集是什么样的？训练使用了什么参数？是从什么开始训练的？跟踪效果怎么差了？

**Reply by wavelet2008** [reply]
大神好啊   训练参数脚本   /pysot/experiments/siamrpn_alex_dwxcorr_16gpu/config.yaml   
META_ARC: "siamrpn_alex_dwxcorr"

BACKBONE:
    TYPE: "alexnet"
    KWARGS:
        width_mult: 1.0   
    PRETRAINED: 'alexnet-bn.pth'
    #PRETRAINED: 'checkpoint_e80.pth'
    TRAIN_LAYERS: ['layer4', 'layer5']
    TRAIN_EPOCH: 5
    LAYERS_LR: 1.0

ADJUST:
    ADJUST: False

RPN:
    TYPE: 'DepthwiseRPN'
    KWARGS:
        anchor_num: 5
        in_channels: 256
        out_channels: 256

MASK:
    MASK: False

ANCHOR:
    STRIDE: 8
    RATIOS: [0.33, 0.5, 1, 2, 3]
    SCALES: [8]
    ANCHOR_NUM: 5

TRACK:
    TYPE: 'SiamRPNTracker'
    PENALTY_K: 0.16
    WINDOW_INFLUENCE: 0.40
    LR: 0.30
    EXEMPLAR_SIZE: 127
    INSTANCE_SIZE: 287
    BASE_SIZE: 0
    CONTEXT_AMOUNT: 0.5

TRAIN:
    EPOCH: 30
    START_EPOCH: 0
    BATCH_SIZE: 128
    BASE_SIZE: 0
    OUTPUT_SIZE: 17
    BASE_LR: 0.005
    CLS_WEIGHT: 10.
    LOC_WEIGHT: 15.2
    RESUME: ''
    #RESUME: 'snapshot/checkpoint_e80.pth'
    LR:
        TYPE: 'log'
        KWARGS:
            start_lr: 0.01
            end_lr: 0.0005
    LR_WARMUP:
        TYPE: 'step'
        EPOCH: 5
        KWARGS:
            start_lr: 0.005
            end_lr: 0.01
            step: 1

DATASET:
   NAMES:
     #- 'VID'
      - 'YOUTUBEBB'
      #- 'COCO'
      #- 'DET'

    TEMPLATE:
        SHIFT: 4
        SCALE: 0.05
        BLUR: 0.0
        FLIP: 0.0
        COLOR: 1.0

    SEARCH:
        SHIFT: 64
        SCALE: 0.18
        BLUR: 0.2
        FLIP: 0.0
        COLOR: 1.0

    NEG: 0.05
    GRAY: 0.0

训练数据 只用 ytbb 中的train0000-train0019    
[2020-11-26 08:48:49,259-rk0-train.py#249] Epoch: [19][1970/3125] lr: 0.001974
	batch_time: 2.171178 (2.176625)	data_time: 1.866371 (1.904743)
	cls_loss: 0.052419 (0.049440)	loc_loss: 0.222597 (0.217926)
	total_loss: 3.907659 (3.806876)	
[2020-11-26 08:48:49,259-rk0-log_helper.py#105] Progress: 58220 / 93750 [62%], Speed: 2.177 s/iter, ETA 0:21:28 (D:H:M)

训练效果：
(pysot) ubuntu@ubuntu-X10SRA:~/pysot$ python tools/demo.py     --config experiments/siamrpn_alex_dwxcorr_16gpu/config.yaml     --snapshot experiments/siamrpn_alex_dwxcorr_16gpu/snapshot/checkpoint_e18.pth --video   demo/21_head_video00035 
[494, 126, 63, 85]   第一帧就错了很离谱
0.92190355
[858, 240, 63, 85]
0.96316814
[909, 65, 63, 85]
0.08671292
[929, 162, 60, 85]
0.9830604
[929, 5, 60, 85]
0.14619353
[929, 5, 60, 85]
0.9291736
[929, 5, 60, 85]
0.9297365
[929, 5, 60, 85]
0.9297072
[929, 5, 60, 85]
0.93052185
[929, 7, 60, 85]
0.92811483
[929, 8, 60, 85]
0.93009555
[929, 12, 60, 85]
0.9249779
[929, 10, 60, 85]
0.9232683
[929, 10, 60, 85]
0.92663795

求小样本训练方法与策略

**Reply by ZhiyuanChen** [reply]
……你至少给一个gif动图吧，给几个框我也没法想象你在跟啥啊
